jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ProjectAccountSettingsManager.kt

```diff
@@ -60,6 +61,16 @@ interface ProjectAccountSettingsManager {
     val activeCredentialProvider: ToolkitCredentialsProvider
         @Throws(CredentialProviderNotFound::class) get
 
+    /**
+     * The underlying AWS account for current active credential provider. Return null if credential provider is not set
+     */
+    val activeAwsAccount: String?
+
+    /**
+     * Return the underlying AWS account for the given credential provider. Return null if the credential provider is invalid
+     */
+    fun awsAccount(credentialProvider: ToolkitCredentialsProvider): String?
```

Comment:
Doesn't feel like this belongs here - all the other fields / methods surface information that's related to the current 'connection' or mutate the connection - this does neither.

Reply:
Make sense. Moved the account cache management to a separate application level service.